cmake/scripts/ethupdate.sh

```diff
@@ -63,49 +132,77 @@ do
 		continue
 	fi
 
-	echo "ERROR: Unrecognized argument \"$arg\".";
+	if [[ $arg == "--no-push" ]]; then
+		NO_PUSH=1
+		continue
+	fi
+
+	if [[ $arg == "--use-ssh" ]]; then
+		USE_SSH=1
+		continue
+	fi
+
+	echo "ETHUPDATE - ERROR: Unrecognized argument \"$arg\".";
 	print_help
 	exit 1
 done
 
 if [[ ${REQUESTED_ARG} != "" ]]; then
-	echo "ERROR: Expected value for the \"${REQUESTED_ARG}\" argument";
+	echo "ETHUPDATE - ERROR: Expected value for the \"${REQUESTED_ARG}\" argument";
 	exit 1
 fi
 
-for project in "${PROJECTS[@]}"
+for repository in "${CLONE_REPOSITORIES[@]}"
 do
-	cd $project >/dev/null 2>/dev/null
+	CLONED_THE_REPO=0
+	cd $repository >/dev/null 2>/dev/null
 	if [[ $? -ne 0 ]]; then
-		echo "Skipping ${project} because directory does not exit";
-		cd $ROOT_DIR
-		continue
+		if [[ $REQUESTED_PROJECT == "" ]]; then
+			echo "ETHUPDATE - INFO: Skipping ${repository} because directory does not exit";
+			cd $ROOT_DIR
+			continue
+		else
+			echo "ETHUPDATE - INFO: Repository ${repository} for requested project ${REQUESTED_PROJECT} did not exist. Cloning ..."
+			get_repo_url $repository
+			git clone $REPO_URL
```

Comment:
if we do not need history, you can add `--depth=1` param. I will be a little faster :)